From 4f0421401fa9046b8f6e56fe5a1572b411584dc2 Mon Sep 17 00:00:00 2001
From: Enrico Butera <ebutera@users.berlios.de>
Date: Wed, 22 Feb 2012 14:28:12 +0100
Subject: [PATCH 15/15] mtd: OneNAND: Fix bufferram management if chip has
 2-planes.

Rebase patch from [1]:

"This patch adds code that was lost when it was applied the commit
5988af2319781bc8e0ce418affec4e09cfa77907, it fix the bufferram management
if chip has 2-planes.

For example, without this patch stressing a jffs2 filesystem using
  bonnie++ -u 0:0 -s 32 -m 16 -r 16

produce various
  Header CRC failed on REF_PRISTINE node at ..."

[1]: http://git.igep.es/?p=pub/scm/linux-omap-2.6.git;a=commit;h=8b7a4c26ca069b010f8ee12c695d83476cfcb722

Signed-off-by: Enrico Butera <ebutera@users.berlios.de>
---
 drivers/mtd/onenand/onenand_base.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/mtd/onenand/onenand_base.c b/drivers/mtd/onenand/onenand_base.c
index a839473..be8f0ae 100644
--- a/drivers/mtd/onenand/onenand_base.c
+++ b/drivers/mtd/onenand/onenand_base.c
@@ -2012,6 +2012,9 @@ static int onenand_write_ops_nolock(struct mtd_info *mtd, loff_t to,
 
 			/* In partial page write we don't update bufferram */
 			onenand_update_bufferram(mtd, to, !ret && !subpage);
+			ONENAND_SET_BUFFERRAM1(this);
+			onenand_update_bufferram(mtd, to + this->writesize, !ret && !subpage);
+
 			if (ret) {
 				printk(KERN_ERR "%s: write failed %d\n",
 					__func__, ret);
-- 
1.7.5.4

